wewatch-client/src/views/swiping/swiping-container.js

```diff
@@ -35,52 +48,53 @@ const SwipingContainer = (props) => {
     })
   }
 
-  useEffect(() => {
-    //initialize room value
-    setRoomId(props.location.state.room);
-  })
-
   //listen to the no match event
   useEffect(() => {
     socket.on('noMatchRedirect', () => {
         console.log("insinde redirect")
         setNoMatch(true)
-  });
-});
+   });
 
+  //listen to match event
+    socket.on('matchRedirect', ({matchedMovieId, matchedMovieData}) => {
+      console.log(matchedMovieId);
+      console.log(matchedMovieData);
+    });
+  });
 
   return (
-    <Container>
+  <Container>
     { noMatch ? <Redirect to='/noMatch'/> : null }
     <button onClick={nomatchtest}>No Match test (will be removed)</button>
     <Row>
       <Col>
-      <StyledDiv flex alignItems="center" marginTop={2}>
-        <CloseIcon
-          style={{ color: theme.colors.red, fontSize: '60px'}}
-          onClick={onClickDislike}
-        />
-        <StyledDiv padding={2}>
-          <MovieDetail
-            title={title}
-            year="2016"
-            lengthOfMovie="2h 28m"
-            rating="8/10"
-            genre="Thriller"
-            imageURL="https://2.bp.blogspot.com/_Iau3R3yMIr4/TMBDGzcvwSI/AAAAAAAACbQ/77grl8TYdK4/s1600/inception1.jpg"
-            description="Inception is a 2010 science fiction action film written and directed by Christopher Nolan, who also produced the film with his wife, Emma Thomas. The film stars Leonardo DiCaprio as a professional thief who steals information by infiltrating the subconscious of his targets."
+      {props.location.state.topTenMovies && (
+        <StyledDiv flex alignItems="center" marginTop={2}>
+          <CloseIcon
+            style={{ color: theme.colors.red, fontSize: '60px'}}
+            onClick={onClickDislike}
+          />
+          <StyledDiv padding={2}>
+            <MovieDetail
+              title={title}
+              year="2016"
+              lengthOfMovie="2h 28m"
+              rating="8/10"
+              genre="Thriller"
+              imageURL={imageURL}
+              description="Inception is a 2010 science fiction action film written and directed by Christopher Nolan, who also produced the film with his wife, Emma Thomas. The film stars Leonardo DiCaprio as a professional thief who steals information by infiltrating the subconscious of his targets."
```

Comment:
I believe that topTenMovies[index] should have a field called "synopsis" that can allow our description here to be dynamic and change with the movies. Would it be possible to add that reference here?

Reply:
right yup! the task of iterating through the movies and making sure that the UI has all details passed through is @varoongupta 's story 1 :)

Reply:
Oh that's right. My mistake, this is more than adequate for stubbed functionality!

Reply:
yup for sure, it was because there was a lag between the movies iterating as the UI was slow to update the image it was showing. I figured I would leave that imageURL and title functionality in here for this PR. Will note for next time!